vsgenstubs4/_vapoursynth.part.pyi

```diff
@@ -426,83 +653,183 @@ class ColorPrimaries(enum.IntEnum):
+PRIMARIES_BT470_M: Literal[ColorPrimaries.PRIMARIES_BT470_M]
+PRIMARIES_BT470_BG: Literal[ColorPrimaries.PRIMARIES_BT470_BG]
+PRIMARIES_ST170_M: Literal[ColorPrimaries.PRIMARIES_ST170_M]
+PRIMARIES_ST240_M: Literal[ColorPrimaries.PRIMARIES_ST240_M]
+PRIMARIES_FILM: Literal[ColorPrimaries.PRIMARIES_FILM]
+PRIMARIES_BT2020: Literal[ColorPrimaries.PRIMARIES_BT2020]
+PRIMARIES_ST428: Literal[ColorPrimaries.PRIMARIES_ST428]
+PRIMARIES_ST431_2: Literal[ColorPrimaries.PRIMARIES_ST431_2]
+PRIMARIES_ST432_1: Literal[ColorPrimaries.PRIMARIES_ST432_1]
+PRIMARIES_EBU3213_E: Literal[ColorPrimaries.PRIMARIES_EBU3213_E]
 
 
 ###
 # VapourSynth Environment SubSystem
+
+
 class EnvironmentData:
-    """
-    Contains the data VapourSynth stores for a specific environment.
-    """
+    def __init__(self) -> NoReturn: ...  # type: ignore[misc]
+
+
+class EnvironmentPolicy:
+    def on_policy_registered(self, special_api: 'EnvironmentPolicyAPI') -> None: ...
+
+    def on_policy_cleared(self) -> None: ...
+
+    @abstractmethod
+    def get_current_environment(self) -> Union[EnvironmentData, None]: ...
+
+    @abstractmethod
+    def set_environment(self, environment: Union[EnvironmentData, None]) -> Union[EnvironmentData, None]: ...
+
+    def is_alive(self, environment: EnvironmentData) -> bool: ...
+
+
+class StandaloneEnvironmentPolicy(EnvironmentPolicy):
+    def __init__(self) -> NoReturn: ...  # type: ignore[misc]
+
+    def _on_log_message(self, level: MessageType, msg: str) -> None: ...
+
+    def on_policy_registered(self, api: 'EnvironmentPolicyAPI') -> None: ...
+
+    def on_policy_cleared(self) -> None: ...
+
+    def get_current_environment(self) -> EnvironmentData: ...
+
+    def set_environment(self, environment: Union[EnvironmentData, None]) -> EnvironmentData: ...
+
+    def is_alive(self, environment: EnvironmentData) -> bool: ...
+
+
+class VSScriptEnvironmentPolicy(EnvironmentPolicy):
+    def __init__(self) -> NoReturn: ...  # type: ignore[misc]
+
+    def on_policy_registered(self, policy_api: 'EnvironmentPolicyAPI') -> None: ...
+
+    def on_policy_cleared(self) -> None: ...
+
+    def get_current_environment(self) -> Union[EnvironmentData, None]: ...
+
+    def set_environment(self, environment: Union[EnvironmentData, None]) -> Union[EnvironmentData, None]: ...
+
+    def is_alive(self, environment: EnvironmentData) -> bool: ...
+
+
+class EnvironmentPolicyAPI:
+    def __init__(self) -> NoReturn: ...  # type: ignore[misc]
+
+    def wrap_environment(self, environment_data: EnvironmentData) -> 'Environment': ...
+
+    def create_environment(self, flags: int = 0) -> EnvironmentData: ...
+
+    def set_logger(self, env: EnvironmentData, logger: Callable[[int, str], None]) -> None: ...
+
+    def destroy_environment(self, env: EnvironmentData) -> None: ...
+
+    def unregister_policy(self) -> None: ...
+
+
+def register_policy(policy: EnvironmentPolicy) -> None:
+    ...
+
+
+if not TYPE_CHECKING:
```

Comment:
Can you tell me when, on a .pyi-file, TYPE_CHECKING is ever `False`?

Reply:
https://github.com/vapoursynth/vsrepo/pull/186#discussion_r973697279